opentelemetry-api/tests/util/test__importlib_metadata.py

```diff
@@ -12,13 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from sys import version_info
 from unittest import TestCase
 
 from opentelemetry.metrics import MeterProvider
 from opentelemetry.util._importlib_metadata import EntryPoint, EntryPoints
 from opentelemetry.util._importlib_metadata import (
     entry_points as importlib_metadata_entry_points,
 )
+from opentelemetry.util._importlib_metadata import version
+
+
+class TestDependencies(TestCase):
+
+    def test_dependencies(self):
+
+        # pylint: disable=import-outside-toplevel
+        # pylint: disable=unused-import
+        # pylint: disable=import-error
+        if version_info.minor < 10:
```

Comment:
I think moving doing two different tests depending on the version of python would be more clear, also using the full version_info to compare

Reply:
No, if we use 2 different tests that are ran or skipped depending on the python version, we will either end up with duplicated code for both tests or testing something different for different python versions and that is exactly what we do not want to do. The idea of this test is that the exact same test is run for every python version so that we are sure that the behavior of our code is the same regardless of the python version. A successful test would mean that our `opentelemetry.util._importlib_metadata` layer is doing its job of abstracting the differences between python versions and yielding a uniform behavior to its users.